test/integration/master/audit_dynamic_test.go

```diff
@@ -0,0 +1,293 @@
+		})
+		require.NoError(t, err, finalErr)
+	})
+
+	t.Run("two sink", func(t *testing.T) {
+		_, err := kubeclient.AuditregistrationV1alpha1().AuditSinks().Create(&sink2)
+		require.NoError(t, err, "failed to create audit sink2")
+		t.Log("created audit sink2")
+
+		var finalErr error
+		err = wait.PollImmediate(100*time.Millisecond, 10*time.Second, func() (bool, error) {
+			// reset event list
+			eventList1.Store(auditinternal.EventList{})
+			eventList2.Store(auditinternal.EventList{})
+
+			// perform configmap ops
+			configMapOperations(t, kubeclient)
+
+			// check for corresponding events in both sinks
+			missing, err := checkForEvents(t, eventList1)
+			if err != nil {
+				finalErr = fmt.Errorf("%v: failed to match all expected events for sink1, events %#v not found", err, missing)
+				return false, nil
+			}
+
+			missing, err = checkForEvents(t, eventList2)
+			if err != nil {
+				finalErr = fmt.Errorf("%v: failed to match all expected events for sink1, events %#v not found", err, missing)
+				return false, nil
+			}
+
+			return true, nil
+		})
+		require.NoError(t, err, finalErr)
+	})
+
+	t.Run("delete sink", func(t *testing.T) {
+		err := kubeclient.AuditregistrationV1alpha1().AuditSinks().Delete("test2", &metav1.DeleteOptions{})
+		require.NoError(t, err, "failed to delete audit sink2")
+		t.Log("deleted audit sink2")
+
+		var finalErr error
+		err = wait.PollImmediate(100*time.Millisecond, 10*time.Second, func() (bool, error) {
+			// reset event lists
+			eventList1.Store(auditinternal.EventList{})
+			eventList2.Store(auditinternal.EventList{})
+
+			// perform configmap ops
+			configMapOperations(t, kubeclient)
+
+			// check for corresponding events in sink1
+			missing, err := checkForEvents(t, eventList1)
+			if err != nil {
+				finalErr = fmt.Errorf("%v: failed to match all expected events for sink1, events %#v not found", err, missing)
+				return false, nil
+			}
+
+			// check that sink2 is empty
+			if len(eventList2.Load().(auditinternal.EventList).Items) != 0 {
+				finalErr = fmt.Errorf("eventList2 should be empty")
+				return false, nil
+			}
+			return true, nil
+		})
+		require.NoError(t, err, finalErr)
+	})
+
+	t.Run("update sink", func(t *testing.T) {
+		sink1, err := kubeclient.AuditregistrationV1alpha1().AuditSinks().Get("test1", metav1.GetOptions{})
+		require.NoError(t, err)
+		sink1.Spec.Webhook.ClientConfig.URL = &server2.URL
+		_, err = kubeclient.AuditregistrationV1alpha1().AuditSinks().Update(sink1)
+		require.NoError(t, err, "failed to update audit sink1")
+		t.Log("updated audit sink1")
+
+		var finalErr error
+		err = wait.PollImmediate(100*time.Millisecond, 10*time.Second, func() (bool, error) {
+			// reset event lists
+			eventList1.Store(auditinternal.EventList{})
+			eventList2.Store(auditinternal.EventList{})
+
+			// perform configmap ops
+			configMapOperations(t, kubeclient)
+
+			// check for corresponding events in sink2
+			missing, err := checkForEvents(t, eventList2)
```

Comment:
do we have already have tests verifying handoff from one sink to another doesn't duplicate or drop any events?

rather than looping and clearing the stores for the sink handoff here, I'd rather see something that:
1. rapidly produces unique events until sink2's store starts receiving events
2. takes the last event from sink1's store and the first from sink2's store, and verifies they are consecutive events
3. polls until sink2's store contains all the events we produced after it's first event
4. verifies no other further events were delivered to sink1